File: 26.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() < 2) return nums.size();
        int subscript = 1;
        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] != nums[i-1]) nums[subscript++] = nums[i];
        }
        return subscript;
    }
};

int main(int argc, char const *argv[])
{
    Solution s;
    vector<int> a = {1,2,2,3,3,4,4};
    cout << s.removeDuplicates(a) << endl;
    return 0;
}

File: 26.py
# class Solution:
#     def removeDuplicates(self, nums: list) -> int:
#         if len(nums) < 2:
#             return len(nums)
#         i = 1
#         while i < len(nums):
#             if nums[i] == nums[i-1]:
#                 nums.pop(i)
#             else:
#                 i += 1
#         return len(nums)

class Solution:
    def removeDuplicates(self, nums) -> int:
        if len(nums)<2:
            return len(nums)
        subscript = 0
        for num in nums:
            if num != nums[subscript]:
                subscript += 1
                nums[subscript] = num
        return subscript + 1

if __name__ == "__main__":
    s = Solution()
    nums = [1,2,2,3,3,3,4]
    print(s.removeDuplicates(nums))
    print(nums)
        